test/rekt/features/broker/control_plane.go

```diff
@@ -627,6 +629,23 @@ func brokerClassIsImmutable(ctx context.Context, t feature.T) {
 	}
 }
 
+func brokerConfigIsImmutable(ctx context.Context, t feature.T) {
+	broker := getBroker(ctx, t)
+
+	broker.Spec = eventingv1.BrokerSpec{
+		Config: &duckv1.KReference{
+			Kind:       "kind",
+			Namespace:  "namespace",
+			Name:       "name",
+			APIVersion: "apiversion",
+		},
+	}
+
+	if _, err := Client(ctx).Brokers.Update(ctx, broker, metav1.UpdateOptions{}); err == nil {
+		t.Errorf("broker.spec.config is mutable")
+	}
+}
```

Comment:
We cannot update a shared object and leave it in a bad state where after we will do new assertions.

I'd recommend to create and update a new one based on the existing one.

The spec also says this is a MUST https://github.com/knative/specs/blob/8e560194c5ca5c290fa9e0003592afd2c423d57e/specs/eventing/control-plane.md?plain=1#L144